src/main/java/de/retest/recheck/report/TestReportFilter.java

```diff
@@ -78,20 +79,21 @@ static StateDifference filter( final StateDifference stateDifference, final Filt
 
 	static List<RootElementDifference> filter( final List<RootElementDifference> rootElementDifferences,
 			final Filter filter ) {
-		final List<RootElementDifference> newRootElementDifferences = new ArrayList<>();
-		for ( final RootElementDifference rootElementDifference : rootElementDifferences ) {
-			newRootElementDifferences.add( filter( rootElementDifference, filter ) );
-		}
-		return newRootElementDifferences;
+		return rootElementDifferences.stream() //
+				.map( rootElementDifference -> filter( rootElementDifference, filter ) ) //
+				.flatMap( newRootElementDifference -> newRootElementDifference.map( Stream::of ) //
+						.orElseGet( Stream::empty ) ) //
```

Comment:
You have this code twice, do you want to extract a method `optionalToStream`?

What is more readable?
```
rootElementDifferences.stream() //
        .map( rootElementDifference -> filter( rootElementDifference, filter ) ) //
        .filter( Optional::isPresent ) //
        .map( Optional::get ) //
        .collect( Collectors.toList );
```

```
rootElementDifferences.stream() //
        .map( rootElementDifference -> filter( rootElementDifference, filter ) ) //
        .flatMap( optionalToStream() ) //
        .collect( Collectors.toList );
```